### PR TITLE
Feature/Rewrite to async

### PR DESCRIPTION
**Pull request by yllfejziu**
**Note**: This is a major update for PovioKitAuth. The old implementation with promises is overridden with async/await.
To be tested more for any edge cases before merging / releasing.

The following social connections need to be tested:

- [x] SignIn with Apple
- [x] SignIn with Google
- [x] SignIn with Facebook
- [x] SignIn with LinkedIn

### CODE REVIEW

**Review by borut-t — APPROVED** [review]
Awesome 🙌

File: Sources/Apple/AppleAuthenticator.swift [hunk 1]
```diff
@@ -34,49 +33,44 @@ public final class AppleAuthenticator: NSObject {
...
+  public func checkAuthentication() async -> Authenticated {
     guard let userId = storage.string(forKey: storageUserIdKey) else {
-      return .value(false)
+      return false
     }
-    
-    return Promise { seal in
+
+    return await withCheckedContinuation { continuation in
       ASAuthorizationAppleIDProvider().getCredentialState(forUserID: userId) { credentialsState, _ in
```
Comment (tonikocjan) [inline Sources/Apple/AppleAuthenticator.swift:68]: Maybe we can consider using the [async version](https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidprovider/3175423-getcredentialstate#discussion) if this method.

Comment (borut-t) [reply Sources/Apple/AppleAuthenticator.swift:68]: Wow, didn't notice the concurrency method. Yes, let's do it. cc @yllfejziu

Comment (yllfejziu) [reply Sources/Apple/AppleAuthenticator.swift]: Nice catch 👍

**Review by borut-t — APPROVED** [review]
👌

File: Sources/Apple/AppleAuthenticator.swift [hunk 2]
```diff
@@ -34,49 +33,41 @@ public final class AppleAuthenticator: NSObject {
...
   
   /// Returns the current authentication state.
   public var isAuthenticated: Authenticated {
     storage.string(forKey: storageUserIdKey) != nil && storage.bool(forKey: storageAuthenticatedKey)
   }
   
-  /// Checks the current auth state and returns the boolean value as promise.
-  public var checkAuthentication: Promise<Authenticated> {
+  /// Checks the current auth state and returns the boolean value asynchronously.
+  public func checkAuthentication() async -> Authenticated {
```
Comment (tonikocjan) [inline Sources/Apple/AppleAuthenticator.swift:62]: Perhaps this function should be throwable and the error handled at the call site.

Comment (borut-t) [reply Sources/Apple/AppleAuthenticator.swift:62]: Let's remove this function completely. We have a mechanism in terms of notification about the revocation.

Comment (yllfejziu) [reply Sources/Apple/AppleAuthenticator.swift]: Removed.

**Review by borut-t — APPROVED** [review]
👌

**Pull request merged**